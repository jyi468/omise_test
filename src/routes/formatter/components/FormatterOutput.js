import React from 'react';

const FormatterOutput = ({output}) => (
    <div className="form-group col-lg-6">
        <label htmlFor="formatterOutput">Output</label>
        <textarea className="form-control formatter-output" id="formatterOutput" rows="20" disabled value={output}/>
    </div>
);

export default FormatterOutput;