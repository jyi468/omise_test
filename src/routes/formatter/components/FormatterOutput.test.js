import React from 'react';
import renderer from 'react-test-renderer';
import FormatterOutput from './FormatterOutput';

it('renders output correctly', () => {
    const output = [
        {
            "id": 10,
            "title": "House",
            "level": 0,
            "children": [
                {
                    "id": 12,
                    "title": "Red Roof",
                    "level": 1,
                    "children": [],
                    "parent_id": 10
                }
            ],
            "parent_id": null
        }
    ];
    const tree = renderer
        .create(<FormatterOutput output={output}/>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});