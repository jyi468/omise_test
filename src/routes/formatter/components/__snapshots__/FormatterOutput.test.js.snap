// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders output correctly 1`] = `
<div
  className="form-group col-lg-6"
>
  <label
    htmlFor="formatterOutput"
  >
    Output
  </label>
  <textarea
    className="form-control formatter-output"
    disabled={true}
    id="formatterOutput"
    rows="20"
    value={
      Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "id": 12,
              "level": 1,
              "parent_id": 10,
              "title": "Red Roof",
            },
          ],
          "id": 10,
          "level": 0,
          "parent_id": null,
          "title": "House",
        },
      ]
    }
  />
</div>
`;
